src/Tree/Entity/Repository/ClosureTreeRepository.php

```diff
@@ -153,14 +153,14 @@ public function childrenQueryBuilder($node = null, $direct = false, $sortByField
             if (is_array($sortByField)) {
                 foreach ($sortByField as $key => $field) {
                     $fieldDirection = strtolower(is_array($direction) ? ($direction[$key] ?? 'asc') : $direction);
-                    if ($meta->hasField($field) && in_array($fieldDirection, ['asc', 'desc'], true)) {
+                    if (($meta->hasField($field) || $meta->isSingleValuedAssociation($field)) && in_array($fieldDirection, ['asc', 'desc'], true)) {
```

Comment:
IMO, we should move the cheaper check first:
```suggestion
                    if (in_array($fieldDirection, ['asc', 'desc'], true) && ($meta->hasField($field) || $meta->isSingleValuedAssociation($field))) {
```

Reply:
Those function calls are not really expensive I think - just checking if array keys exist and doing bitwise operation.
I've moved the `strtolower()` call in the `if`, so it's now not always called, but when it is, it would probably be the most expensive call, so I've left the order as it is.